### DISCUSSION TITLE
Filter: checking array emptiness

### DISCUSSION THREAD

**Discussion by soulseekah** [directus/directus#18039] [Feature Requests]
### Summary

There is currently no way to check if an array is empty or not.

`_empty` does not work as per https://github.com/directus/directus/issues/18028

while `"_eq": []` is not supported

### Basic Example

Create a flow, Read step that returns [] when nothing is found, Condition that checks `{"$last": {"_empty": true}}` or `{"$last":{"_eq": []}}`. First option expects `''`, `false` or `null`. Second option errors out as unsupported operand.

### Motivation

Checking if an array is empty or not is a quite important and foundational construct to have available in any system.

### Detailed Design

`_empty` and `_nempty` should treat an empty `[]` as falsy, despite them being always truthy.

### Must Have:

- `_empty` checks for `length` of 0 on array typed values
- `_nempty` checks for `length` of >= 1 on array typed values

### Should Have:

n/a

### Could Have:

n/a

### Won't Have:

n/a

### Drawbacks

There are no known drawbacks to supporting arrays in `_empty` and `_nempty`.

### Alternatives

Writing a custom script that checks `.length` is a usable approach, before applying a condition.

![image](https://user-images.githubusercontent.com/685880/229561163-2379084a-1527-48b4-a771-099ab7aeb86c.png)

### Adoption Strategy

`_empty` will start supporting arrays. A notice will be posted on the release notes to notify people of the possible breaking change

### Unresolved Questions

n/a

**Comment by rijkvanzanten** [comment]
What would happen for an array containing "empty" items, like `['', false, null]` ?

**Reply by soulseekah** [reply]
That array is not empty. It has 3 items in it. `.length = 3`, ergo not empty.

**Comment by rijkvanzanten** [comment]
This feature request has been accepted and is queued to be implemented! You can follow along with the progress here:

https://github.com/directus/directus/issues/18042
